sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DefaultStepFactory.java

```diff
@@ -162,7 +162,7 @@ private boolean hasReachedGoalState(PodInstance podInstance, Protos.TaskInfo tas
                 default:
                     return false;
             }
-        } else if (goalState.equals(GoalState.FINISHED)) {
+        } else if (goalState.equals(GoalState.ONCE) || goalState.equals(GoalState.FINISH)) {
```

Comment:
This logic is actually incorrect, but no more incorrect than it already is.  See: https://jira.mesosphere.com/browse/INFINITY-2837